service/chain/follower.go

```diff
@@ -0,0 +1,316 @@
+		return nil, fmt.Errorf("could not lookup collections: %w", err)
+	}
+
+	collections := make([]*flow.LightCollection, 0, len(collIDs))
+	for _, collID := range collIDs {
+		var collection flow.LightCollection
+		err := f.db.View(operation.RetrieveCollection(collID, &collection))
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve collection (%x): %w", collID, err)
+		}
+		collections = append(collections, &collection)
+	}
+
+	return collections, nil
+}
+
+func (f *Follower) Guarantees(height uint64) ([]*flow.CollectionGuarantee, error) {
+
+	blockID, err := f.block(height)
+	if errors.Is(err, dps.ErrTimeout) {
+		return nil, dps.ErrTimeout
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not get block for height: %w", err)
+	}
+
+	var collIDs []flow.Identifier
+	err = f.db.View(operation.LookupPayloadGuarantees(blockID, &collIDs))
+	if err != nil {
+		return nil, fmt.Errorf("could not lookup collections: %w", err)
+	}
+
+	guarantees := make([]*flow.CollectionGuarantee, 0, len(collIDs))
+	for _, collID := range collIDs {
+		var guarantee flow.CollectionGuarantee
+		err := f.db.View(operation.RetrieveGuarantee(collID, &guarantee))
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve guarantee (%x): %w", collID, err)
+		}
+		guarantees = append(guarantees, &guarantee)
+	}
+
+	return guarantees, nil
+}
+
+func (f *Follower) Transactions(height uint64) ([]*flow.TransactionBody, error) {
+
+	blockID, err := f.block(height)
+	if errors.Is(err, dps.ErrTimeout) {
+		return nil, dps.ErrTimeout
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not get block for height: %w", err)
+	}
+
+	var collIDs []flow.Identifier
+	err = f.db.View(operation.LookupPayloadGuarantees(blockID, &collIDs))
+	if err != nil {
+		return nil, fmt.Errorf("could not lookup collections: %w", err)
+	}
+
+	var transactions []*flow.TransactionBody
+	for _, collID := range collIDs {
+		var collection flow.LightCollection
+		err := f.db.View(operation.RetrieveCollection(collID, &collection))
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve collection (%x): %w", collID, err)
+		}
+		for _, txID := range collection.Transactions {
+			var transaction flow.TransactionBody
+			err := f.db.View(operation.RetrieveTransaction(txID, &transaction))
+			if err != nil {
+				return nil, fmt.Errorf("could not retrieve transaction (%x): %w", txID, err)
+			}
+			transactions = append(transactions, &transaction)
+		}
+	}
+
+	return transactions, nil
+}
+
+func (f *Follower) Results(height uint64) ([]*flow.TransactionResult, error) {
+	blockID, err := f.block(height)
+	if errors.Is(err, dps.ErrTimeout) {
+		return nil, dps.ErrTimeout
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not get block for height: %w", err)
+	}
+
+	var results []flow.TransactionResult
+	err = f.db.View(operation.LookupTransactionResultsByBlockID(blockID, &results))
+	if err != nil {
+		return nil, fmt.Errorf("could not lookup transaction results: %w", err)
+	}
```

Comment:
These are also only available in the execution node feed, not in the state of the follower.